### DISCUSSION TITLE
Exception when using `testclient.TestClient` from within an async test

### DISCUSSION THREAD

**Discussion by uriva** [fastapi/fastapi#7347] [Questions] [answered]
Is `testclient.TestClient` supposed to work within a test that has some async in it?

This is how my test looks like:

```python
async def test_something():
    some_input = await compute_input()
    app = make_app(some_input)
    assert testclient.TestClient(app).post("/some_endpoint", json={}) == {}
```
I'm running it via `pytest` and using `pytestmark = pytest.mark.asyncio`

And I'm getting this exception:

> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> /usr/local/lib/python3.8/site-packages/requests/sessions.py:578: in post
>     return self.request('POST', url, data=data, json=json, **kwargs)
> /usr/local/lib/python3.8/site-packages/starlette/testclient.py:413: in request
>     return super().request(
> /usr/local/lib/python3.8/site-packages/requests/sessions.py:530: in request
>     resp = self.send(prep, **send_kwargs)
> /usr/local/lib/python3.8/site-packages/requests/sessions.py:643: in send
>     r = adapter.send(request, **kwargs)
> /usr/local/lib/python3.8/site-packages/starlette/testclient.py:243: in send
>     raise exc from None
> /usr/local/lib/python3.8/site-packages/starlette/testclient.py:240: in send
>     loop.run_until_complete(self.app(scope, receive, send))
> /usr/local/lib/python3.8/asyncio/base_events.py:592: in run_until_complete
>     self._check_running()
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
> self = <_UnixSelectorEventLoop running=False closed=False debug=False>
>     def _check_running(self):
>         if self.is_running():
> >           raise RuntimeError('This event loop is already running')
> E           RuntimeError: This event loop is already running
> /usr/local/lib/python3.8/asyncio/base_events.py:552: RuntimeError
> ----------------------------- Captured stdout call -----------------------------
>

**Comment by jahhhsueihh** [comment]
في أربعاء، 15 يوليو، 2020 في 5:15 ص، كتب uriva <notifications@github.com>:

**Comment by jahhhsueihh** [comment]
في أربعاء، 15 يوليو، 2020 في 5:15 ص، كتب عبدالله السعدي <
3bdulla.alsaadi@gmail.com>:

**Comment by uriva** [comment, marked as answer]
Duplicated with #1273
